core/src/apps/cardano/layout.py

```diff
@@ -64,17 +96,126 @@ def is_printable_ascii_bytestring(bytestr: bytes) -> bool:
+    script: CardanoNativeScript,
+    indices: list[int] = [],
+) -> None:
+    indices_str = ".".join([str(i) for i in indices])
+
+    script_type_name_suffix = ""
+    if script.type == CardanoNativeScriptType.PUB_KEY:
+        if script.key_path:
+            script_type_name_suffix = "path"
+        elif script.key_hash:
+            script_type_name_suffix = "hash"
+
+    props: list[PropertyType] = [
+        (
+            "Script%s - %s %s:"
+            % (
+                (" " + indices_str if indices_str else ""),
+                SCRIPT_TYPE_NAMES[script.type],
+                script_type_name_suffix,
+            ),
+            None,
+        )
+    ]
+
+    if script.type == CardanoNativeScriptType.PUB_KEY:
+        assert script.key_hash is not None or script.key_path  # validate_script
+        if script.key_hash:
+            props.append((None, format_key_hash(script.key_hash, True)))
+        elif script.key_path:
+            props.append((address_n_to_str(script.key_path), None))
+    elif script.type == CardanoNativeScriptType.N_OF_K:
+        assert script.required_signatures_count is not None  # validate_script
+        props.append(
+            (
+                "Requires %s out of %s signatures."
+                % (script.required_signatures_count, len(script.scripts)),
+                None,
+            )
+        )
+    elif script.type == CardanoNativeScriptType.INVALID_BEFORE:
+        assert script.invalid_before is not None  # validate_script
+        props.append((str(script.invalid_before), None))
+    elif script.type == CardanoNativeScriptType.INVALID_HEREAFTER:
+        assert script.invalid_hereafter is not None  # validate_script
+        props.append((str(script.invalid_hereafter), None))
+
+    if script.type in (
+        CardanoNativeScriptType.ALL,
+        CardanoNativeScriptType.ANY,
+        CardanoNativeScriptType.N_OF_K,
+    ):
+        assert script.scripts  # validate_script
+        props.append(("Contains %i nested scripts." % len(script.scripts), None))
+
+    await confirm_properties(
+        ctx,
+        "verify_script",
+        title="Verify script",
+        props=props,
+        br_code=ButtonRequestType.Other,
+    )
+
+    for i, sub_script in enumerate(script.scripts):
+        await show_native_script(ctx, sub_script, indices + [(i + 1)])
+
+
+async def show_script_hash(
+    ctx: wire.Context,
+    script_hash: bytes,
+    display_format: CardanoNativeScriptHashDisplayFormat,
+) -> None:
+    assert display_format in (
+        CardanoNativeScriptHashDisplayFormat.BECH32,
+        CardanoNativeScriptHashDisplayFormat.POLICY_ID,
+    )
+
+    props: list[PropertyType] = []
+    if display_format == CardanoNativeScriptHashDisplayFormat.BECH32:
+        props = [("Script hash:", format_script_hash(script_hash))]
+    elif display_format == CardanoNativeScriptHashDisplayFormat.POLICY_ID:
+        props = [("Policy ID:", script_hash)]
+
+    await confirm_properties(
```

Comment:
Right. This should either be `confirm_blob` or `confirm_properties` if `display_format` is `POLICY_ID` or `BECH32` respectively.

Fixed in https://github.com/trezor/trezor-firmware/pull/1772/commits/6819215aa240da1e8f3e7749c8a430dd191198d9#diff-718c35fab848e1ee6f5e2a387c71b0a0ea8d84039fedce530cb9f0584c164495R187

Reply:
why should this _not_ be `confirm_blob` if the format is bech32?

Reply:
Bech32 is meant as a human-readable format, not a binary blob. (Furthermore, we show bech32-encoded strings everywhere as text/props, so it makes sense to be consistent here.)

Reply:
> Bech32 is meant as a human-readable format

Except it's really not that. "human-readable" are plain texts, numbers, dates, etc. Per the docstring, `confirm_blob` is "Applicable for public keys, signatures, hashes. In general, any kind of data that is not human-readable, and can be wrapped at any character." (plus `confirm_address` is internally an instance of `confirm_blob`.

> Furthermore, we show bech32-encoded strings everywhere as text/props, so it makes sense to be consistent here

That's only because we currently don't have the option to select rendering mode per prop.

---

I'm OK with leaving it as is if you want consistency for now, but be advised that we'll be overhauling the rendering in the near future.

Reply:
Okay, thanks for clarification.

I agree with leaving this consistent with other bech32 occurrences for now and taking another look at this once the displaying api changes.